### PR TITLE
Support rand(Gaussian{Vector{T}})

### PR DESCRIPTION
**Pull request by cstjean**

### CODE REVIEW

File: src/GaussianDistributions.jl [hunk 1]
```diff
@@ -59,9 +59,11 @@ whiten(Σ::UniformScaling, z) = z/sqrt(Σ.λ)
 sqmahal(P::Gaussian, x) = norm_sqr(whiten(P.Σ,x - P.μ))
 
 rand(P::Gaussian) = P.μ + chol(P.Σ)'*randn(typeof(P.μ))
-rand(P::Gaussian{Vector}) = P.μ + chol(P.Σ)'*randn(T, length(P.μ))
+rand(P::Gaussian{V}) where V<:AbstractVector{T} where T =
```
Comment (mschauer) [inline src/GaussianDistributions.jl]: `V<:AbstractVector{T}` would catch `SVector`'s, but those have to call `randn(typeof(P.μ))`

Comment (cstjean) [reply src/GaussianDistributions.jl]: Oh, that's neat... I'll fix it.

**Comment by cstjean** [comment]
With this PR, `rand(::Gaussian{Vector{T}}, Tuple{Int})` returns a `Array{T, 2}` (i.e. one big matrix). Perhaps it should return a `Vector{Vector{T}}` instead? That would be ugly and wasteful, but it would be more symmetric with `SVector` handling, and perhaps would require less special-casing in `Kalman/src/generate.jl`? Any opinion on that?

Or maybe handling both `Vector` and `SVector` is a pain, and we just focus our efforts on `SVector`?

**Comment by mschauer** [comment]
Actually I like to think of an mxn array or `Array{T, 2}` as being the counterpart  to a n-vector of static m-vectors or `Vector{SVector}`. By https://github.com/mschauer/Kalman.jl/blob/master/src/ellipse.jl 
```
A[.., i]
```
in both cases refers to the `i`'s column vector.

**Comment by cstjean** [comment]
Ok. I fixed the issue. Let me know if there's anything else.

File: src/GaussianDistributions.jl [hunk 2]
```diff
@@ -75,14 +77,15 @@ function rand(RNG, P::Gaussian{T}, dims) where {T}
     X
 end
 
-function rand(P::Gaussian{T}, dims) where {T}
```
Comment (mschauer) [inline src/GaussianDistributions.jl:78]: We also still need this one.

Comment (cstjean) [reply src/GaussianDistributions.jl:78]: It's handled by `rand(P::Gaussian, dims) = rand(Base.GLOBAL_RNG, P, dims)`

**Comment by mschauer** [comment]
Great, thank you.

**Pull request merged**